web/src/auth/LoginPage.js

```diff
@@ -871,9 +904,36 @@ class LoginPage extends React.Component {
   renderMethodChoiceBox() {
     const application = this.getApplicationObj();
     const items = [];
-    application.enablePassword ? items.push({label: i18next.t("general:Password"), key: "password"}) : null;
-    application.enableCodeSignin ? items.push({label: i18next.t("login:Verification code"), key: "verificationCode"}) : null;
-    application.enableWebAuthn ? items.push({label: i18next.t("login:WebAuthn"), key: "webAuthn"}) : null;
+
+    for (const index in application.signinMethods) {
+      if (application.signinMethods[index].name === application.signinMethods[index].displayName) {
```

Comment:
Define a variable `displayName` out of the if branch, don't need to repeat code here

Reply:
Refactored using map